lib/api/CFieldDataCategorizer.cc

```diff
@@ -266,33 +230,33 @@ CFieldDataCategorizer::categorizerPtrForKey(const std::string& partitionFieldVal
         return nullptr;
     }
 
+    CCategoryIdMapper::TCategoryIdMapperPtr idMapper;
+    if (m_PartitionFieldName.empty()) {
+        idMapper = std::make_shared<CNoopCategoryIdMapper>();
+    } else {
+        idMapper = std::make_shared<CPerPartitionCategoryIdMapper>(
+            partitionFieldValue, [this]() { return this->nextGlobalId(); });
+    }
+
     // TODO - if we ever have more than one data categorizer class, this
     // should be replaced with a factory
-    auto ptr = std::make_shared<TTokenListDataCategorizerKeepsFields>(
+    auto localCategorizer = std::make_shared<TTokenListDataCategorizerKeepsFields>(
```

Comment:
Isn't their a one-to-one relationship between `localCategorizer` and `globalCategorizer`, i.e. I'd expect this to be a `unique_ptr`.